File: main/src/ui/java/freedom/app/ddns/DdnsServiceType.kt
package freedom.app.ddns

enum class DdnsServiceType(
    val displayName: String,
    val field1Label: String,
    val field2Label: String,
    val field3Label: String?,
    val websiteUrl: String = ""
) {
    NOIP("No-IP", "Hostname", "Username", "Password", "https://www.noip.com"),
    DUCKDNS("DuckDNS", "Domain(s)", "Token", null, "https://www.duckdns.org"),
    DYNU("Dynu", "Hostname", "Username", "Password", "https://www.dynu.com"),
    CLOUDFLARE("Cloudflare", "API Token", "Zone ID", "Record ID", "https://www.cloudflare.com"),
    NAMECHEAP("Namecheap", "Host", "Domain", "DDNS Password", "https://www.namecheap.com"),
    DESEC("deSEC", "Hostname", "E-mail", "Password", "https://desec.io"),
    DYNV6("dynv6", "Hostname", "Token", null, "https://dynv6.com"),
    YDNS("YDNS", "Hostname", "E-mail", "Password", "https://ydns.io"),
    IPV64("IPv64", "Hostname", "API Key", null, "https://ipv64.net"),
    FREEDNS("FreeDNS", "Hostname", "Update Token", null, "https://freedns.afraid.org")
}

File: ios/Freedom/Core/DDNS/DdnsServiceType.swift
// ═════════════════════════════════════════════════════════════════════════════
//  DdnsServiceType — Enum for 10 DDNS providers with field labels
// ═════════════════════════════════════════════════════════════════════════════

import Foundation

enum DdnsServiceType: String, Codable, CaseIterable, Identifiable {
    case noip       = "NOIP"
    case duckdns    = "DUCKDNS"
    case dynu       = "DYNU"
    case cloudflare = "CLOUDFLARE"
    case namecheap  = "NAMECHEAP"
    case desec      = "DESEC"
    case dynv6      = "DYNV6"
    case ydns       = "YDNS"
    case ipv64      = "IPV64"
    case freedns    = "FREEDNS"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .noip:       return "No-IP"
        case .duckdns:    return "DuckDNS"
        case .dynu:       return "Dynu"
        case .cloudflare: return "Cloudflare"
        case .namecheap:  return "Namecheap"
        case .desec:      return "deSEC"
        case .dynv6:      return "dynv6"
        case .ydns:       return "YDNS"
        case .ipv64:      return "IPv64"
        case .freedns:    return "FreeDNS"
        }
    }

    var field1Label: String {
        switch self {
        case .noip:       return "Hostname"
        case .duckdns:    return "Subdomain"
        case .dynu:       return "Hostname"
        case .cloudflare: return "API Token"
        case .namecheap:  return "Host"
        case .desec:      return "Hostname"
        case .dynv6:      return "Hostname"
        case .ydns:       return "Hostname"
        case .ipv64:      return "Hostname"
        case .freedns:    return "Hostname"
        }
    }

    var field2Label: String {
        switch self {
        case .noip:       return "Username"
        case .duckdns:    return "Token"
        case .dynu:       return "Username"
        case .cloudflare: return "Zone ID"
        case .namecheap:  return "Domain"
        case .desec:      return "Email"
        case .dynv6:      return "Token"
        case .ydns:       return "Email"
        case .ipv64:      return "API Key"
        case .freedns:    return "Update Token"
        }
    }

    var field3Label: String? {
        switch self {
        case .noip:       return "Password"
        case .dynu:       return "Password"
        case .cloudflare: return "Record ID"
        case .namecheap:  return "DDNS Password"
        case .desec:      return "Password"
        case .ydns:       return "Password"
        default:          return nil
        }
    }

    /// Whether this provider supports automated registration via API.
    var isAutomatable: Bool {
        switch self {
        case .duckdns, .desec, .dynv6, .ydns, .ipv64, .dynu: return true
        default: return false
        }
    }
}
